specs/NestedFrame.md

```diff
@@ -0,0 +1,268 @@
+CoreWebView2Frame.FrameCreated API
+===
+
+# Background
+At present, WebView2 enables developers to track only first-level
+iframes, which are the direct child iframes of the main frame.
+However, we see that WebView2 customers want to manage nested
+iframes, such as recording the navigation history for a nested
+iframe. To address this, we will introduce the
+`CoreWebView2Frame.FrameCreated` API. This new API will allow
+developers to subscribe to the nested iframe creation event,
+giving them access to all properties, methods, and events of
+[CoreWebView2Frame](https://learn.microsoft.com/dotnet/api/microsoft.web.webview2.core.corewebview2frame)
+for the nested iframe.
+
+To prevent unnecessary performance implication, WebView2 does
+not track any nested iframes by default. It only tracks a nested
+iframe if its parent iframe(`CoreWebView2Frame`) has subscribed
+to the `CoreWebView2Frame.FrameCreated` API. For a page with
+multi-level iframes, developers can choose to track only the
+main page and first-level iframes (the default behavior), a
+partial WebView2 frames tree with specific iframes of interest,
+or the full WebView2 frames tree.
+
+# Examples
+### C++ Sample
+```cpp
+EventRegistrationToken m_frameCreatedToken = {};
+void PostFrameCreatedEventMessage(wil::com_ptr<ICoreWebView2Frame> webviewFrame);
+
+void ScenarioWebViewEventMonitor::InitializeEventView(ICoreWebView2* webviewEventView)
+{
+    auto webviewEventView4 = webviewEventView.try_query<ICoreWebView2_4>();
+    webviewEventView4->add_FrameCreated(
+            Callback<ICoreWebView2FrameCreatedEventHandler>(
+                [this](ICoreWebView2* sender, ICoreWebView2FrameCreatedEventArgs* args)
+                    -> HRESULT {
+                    wil::com_ptr<ICoreWebView2Frame> webviewFrame;
+                    CHECK_FAILURE(args->get_Frame(&webviewFrame));
+                    // Track first-level webview frame events.
+                    InitializeFrameEventView(webviewFrame);
+                    PostFrameCreatedEventMessage(webviewFrame);
+                    return S_OK;
+                })
+                .Get(),
+            &m_frameCreatedToken);
+}
+
+// Track the creation, destruction, and navigation events of webview frames.
+void ScenarioWebViewEventMonitor::InitializeFrameEventView(
+    wil::com_ptr<ICoreWebView2Frame> webviewFrame) {
+    auto frame7 = webviewFrame.try_query<ICoreWebView2Frame7>();
+    if (frame7)
+    {
+        //! [AddFrameCreated]
+        frame7->add_FrameCreated(
+            Callback<ICoreWebView2FrameNestedFrameCreatedEventHandler>(

```

Comment:
Strictly by our usual rules of `I{interface name}{event name}EventHandler` this would be `ICoreWebView2FrameFrameCreatedEventHandler`. If we think it looks weird then perhaps `ICoreWebView2FrameChildFrameCreatedEventHandler`